lib/java_buildpack/framework/introscope_agent.rb

```diff
@@ -80,12 +66,27 @@ def agent_jar
         @droplet.sandbox + 'Agent.jar'
       end
 
-      def agent_manager(credentials)
-        agent_manager = ssl?(credentials) ? 'https://' : 'http://'
-        agent_manager += host_name(credentials)
+      def add_url(credentials, java_opts)
+        agent_manager = url(credentials)
 
-        port = port(credentials)
-        port ? "#{agent_manager}:#{port}" : agent_manager
+        host, port, socket_factory = parse_url(agent_manager)
+        java_opts.add_system_property('agentManager.url.1', agent_manager)
+        java_opts.add_system_property('introscope.agent.enterprisemanager.transport.tcp.host.DEFAULT', host)
```

Comment:
What's the point of setting both the `host` and `agentManager.url.1`?  This doesn't seem particularly different than the original situation where we were asked for the host, port, and ssl and put them together to make the URI.  Assembling a URI from it's components is much more reliable that trying to split it, so if you still need the component parts, then we should be asking for them independently.